experanto/datasets.py

```diff
@@ -0,0 +1,448 @@
+        self._valid_screen_times = self._full_valid_sample_times[::self.sample_stride]
+        self.transforms = self.initialize_transforms()
+
+    def _read_trials(self) -> None:
+        screen = self._experiment.devices["screen"]
+        self._trials = [t for t in screen.trials]
+        start_idx = np.array([t.first_frame_idx for t in self._trials])
+        self._start_times = screen.timestamps[start_idx]
+        self._end_times = np.append(screen.timestamps[start_idx[1:]], np.inf)
+        self.meta_conditions = {}
+        for k in self.modality_config.screen.valid_condition.keys():
+            self.meta_conditions[k] = [t.get_meta(k) if t.get_meta(k) is not None else "blank" for t in self._trials]
+
+    def initialize_statistics(self) -> None:
+        self._statistics = {}
+        for device_name in self.device_names:
+            self._statistics[device_name] = {}
+            # If modality should be normalized, load respective statistics from file.
+            if self.modality_config[device_name].transforms.get("normalization", False):
+                mode = self.modality_config[device_name].transforms.normalization
+                assert mode in ['standardize', 'normalize']
+                means = np.load(self._experiment.devices[device_name].root_folder / "meta/means.npy")
+                stds = np.load(self._experiment.devices[device_name].root_folder / "meta/stds.npy")
+                if mode == 'standardize':
+                    # If modality should only be standarized, set means to 0.
+                    means = np.zeros_like(means)
+
+                self._statistics[device_name]["mean"] = means.reshape(1, -1)  # (n, 1) -> (1, n) for broadcasting in __get_item__
+                self._statistics[device_name]["std"] = stds.reshape(1, -1)  # same as above
+
+    def initialize_transforms(self):
+        transforms = {}
+        for device_name in self.device_names:
+            if device_name == "screen":
+                # for the screen, we'll utilize torchvision.transforms
+                add_channel = Lambda(lambda x: torch.from_numpy(x[:, None, ...]) if len(x.shape) == 3 else torch.from_numpy(x))
+                transform_list = [v for v in self.modality_config.screen.transforms.values()]
+                transform_list.insert(0, add_channel)
+                transforms[device_name] = Compose(transform_list)
+            else:
+                if self.modality_config[device_name].transforms:
+                    transforms[device_name] = Compose([
+                        ToTensor(),
+                        torchvision.transforms.Normalize(self._statistics[device_name]["mean"], self._statistics[device_name]["std"])])
+                else:
+                    transforms[device_name] = Compose([
+                        ToTensor()])
+        return transforms
+
+    def get_sample_in_meta_condition(self) -> dict:
+        """
+        iterates through all
```

Comment:
it took me a bit of time to get through it, so I would suggest to add a bit more info in the doc strings, see the suggestion below
```suggestion
        iterates through all stimuli, selects the ones which match the meta conditions (tiers or stimuli types) and creates a mask to select the correct times using `self._screen_sample_times` as the clock/reference times
```

Reply:
Yes, I'm very sorry that I commented so sparsely. Will commit yours and add more.